app/_hub/kong-inc/ai-proxy/how-to/_sdk-usage.md

```diff
@@ -0,0 +1,257 @@
+---
+title: SDK Usage
+nav_title: Use Programmatic SDKs with AI Proxy
+minimum_version: 3.7.x
+---
+
+You can use an OpenAI-compatible SDK with the AI Proxy plugin in multiple ways, depending on the required use case.
+
+| You want to...       | Then use...                   |
+| -------------------- | -------------------------------------------------- |
+| Allow the user to select their target model, based on some header or request parameter | [OpenAI SDK with multiple models on the same provider](#use-openai-sdk-with-multiple-models-on-the-same-provider) |
+| Proxy the same request to an LLM provider of the user's choosing | [OpenAI SDK with multiple providers](#use-openai-sdk-with-multiple-providers) |
+| Use the OpenAI SDK for Azure, and allow the user to choose the Azure Deployment ID | [Multiple Azure OpenAI deployments on one route](#multiple-azureopenai-deployments-on-one-route) |
+| Proxy an unsupported model, like Whisper-2. | [OpenAI-compatible SDK for unsupported models](#use-an-unsupported-model)               |
+
+
+## Templated Model Parameters
+
+The plugin enables you to substitute values in the `config.model.name` and/or any `config.model.options.*` field,
+with specific placeholders, similar to those in the [Request Transformer Advanced](/hub/kong-inc/request-transformer-advanced/)
+templating system.
+
+The available templated parameters are:
+
+* `$(headers.name)`
+* `$(uri_captures.name)`
+* `$(query_params.name)`
+
+`name` is either the header name, URI named capture (in the route path), or the query parameter name, respectively.
+
+## Use case examples
+### Use OpenAI SDK with multiple models on the same provider
+
+To read the desired model from the user, rather than hard coding it into the plugin config for each route,
+you can read it from a path parameter, for example:
+
+```yaml
+- name: openai-chat
+  paths:
+  - "~/(?<model>[^#?/]+)/chat/completions$"
+  methods:
+  - POST
+  plugins:
+  - name: ai-proxy
+    config:
+      route_type: "llm/v1/chat"
+      auth:
+        header_name: "Authorization"
+        header_value: "{vault://env/OPENAI_AUTH_HEADER}"
+      logging:
+        log_statistics: true
+        log_payloads: false
+      model:
+        provider: "openai"
+        name: "$(uri_captures.model)"
+```
+
+You can now target two different models using an OpenAI-compatible SDK:
+
+```python
+client = OpenAI(
+  base_url="http://localhost:8000/gpt-4"
+)
+```
+
+The Python SDK (in OpenAI standard mode) will fill in the URL with `http://localhost:8000/gpt-4/chat/completions`. {{site.base_gateway}} recognizes the model as "gpt-4" and the route appropriately.
+
+### Using OpenAI SDK with multiple providers
+
+To use the same OpenAI SDK but with multiple LLM providers, Kong AI Proxy can arbitrate this for you.
```

Comment:
```suggestion
Kong AI Proxy can help you arbitrate using the same OpenAI SDK with multiple LLM providers.
```
Something like this? I don't love the word arbitrate here, but I'm going to assume it's helpful in a technical sense.